### PR TITLE
docs: update pr template with checklist

### PR DESCRIPTION
**Pull request by codebytere**
This PR updates the Pull Request template so that contributors see a clearer checklist of items they need to complete in order to have a given PR reviewed and landed.

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
The idea is stolen from Node, isn't it? =)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,7 +2,14 @@
...
--->
\ No newline at end of file
+-->
+
+##### Checklist
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+
+- [ ] `npm test` passes
+- [ ] `npm run lint` passes
+- [ ] tests are included
```
Comment (nornagon) [inline .github/PULL_REQUEST_TEMPLATE.md]: do we have links to guidelines for how to write tests?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
+- [ ] documentation is changed or added
```
Comment (nornagon) [inline .github/PULL_REQUEST_TEMPLATE.md]: perhaps 'relevant documentation'?

**Comment by codebytere** [comment]
@alexeykuzmin yes indeed 😁

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (codebytere) [reply .github/PULL_REQUEST_TEMPLATE.md]: @nornagon i think we do, let me check

**Comment by codebytere** [comment]
@jkleinsc good idea, i was just concerned about being too pedantic

how about something like 
` - [  ] PR description included and relevant parties cc'd`?

**Comment by ckerr** [comment]
:+1: but s/relevant parties/stakeholders/

**Comment by sofianguy** [comment]
@codebytere I don't think it's too much to ask for a description and parties cc'd

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -2,7 +2,15 @@
...
+
+##### Checklist
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+
+- [ ] PR description included and stakeholders cc'd
+- [ ] `npm test` passes
+- [ ] `npm run lint` passes
+- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
+- [ ] relevant documentation is changed or added
+- [ ] commit message follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
```
Comment (zeke) [inline .github/PULL_REQUEST_TEMPLATE.md]: commit message follows -> commit messages or PR title follow

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -2,7 +2,15 @@
...
--->
\ No newline at end of file
+-->
+
+##### Checklist
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+
+- [ ] PR description included and stakeholders cc'd
+- [ ] `npm test` passes
+- [ ] `npm run lint` passes
```
Comment (zeke) [inline .github/PULL_REQUEST_TEMPLATE.md]: If `test` is passing, then `lint` is also necessarily passing. Maybe these should be reversed?

Comment (codebytere) [reply .github/PULL_REQUEST_TEMPLATE.md]: i can remove lint then i think

**Comment by alexeykuzmin** [comment]
> and relevant parties cc'd

I agree to @sofianguy, all required reviewers will be added automatically, all others are, by definition, optional )

**Pull request merged**